alf/algorithms/data_transformer.py

```diff
@@ -131,7 +131,7 @@ def __init__(self, data_transformer_ctors, observation_spec):
                 assert not has_non_frame_stacker, (
                     "FrameStacker need to be the "
                     "first data transformers if it is used.")
-            else:
+            elif not isinstance(obs_trans, HindsightExperienceTransformer):
```

Comment:
What if FrameStacker and Hindsight are used at the same time? Do we need some special check?

Reply:
Good point.  Reverted this line.

We should just throw error.  We haven't thought through this case yet.

Having Hindsight before FrameStack would not work, because FrameStack would try to retrieve data from ReplayBuffer directly.  Having Hindsight after FrameStack, then we need to stack future achieved_goal steps from ReplayBuffer to form stacked desired_goal.